### PR TITLE
[HOSTING-525] Add permissions for kubernetes-credentials-provider

### PR DESCRIPTION
**Pull request by jtnord**
For some reason the permissions were not set on artifactory as a result
of the hosting process request, so filing manually...

# Description

https://issues.jenkins-ci.org/browse/HOSTING-525
https://github.com/jenkinsci/kubernetes-credentials-provider-plugin

# Submitter checklist for changing permissions

### Always

- [x] Add link to plugin/component Git repository in description above

### For a newly hosted plugin only

- [x] Add link to resolved HOSTING issue in description above

### For a new permissions file only

- [x] Make sure the file is created in `permissions/` directory
- [x] `artifactId` (pom.xml) is used for `name` (permissions YAML file).
- [x] [`groupId` / `artifactId` (pom.xml) are correctly represented in `path` (permissions YAML file)](https://github.com/jenkins-infra/repository-permissions-updater/#managing-permissions)
- [x] Check that the file is named `plugin-${artifactId}.yml` for plugins

### When adding new uploaders (this includes newly created permissions files)

- [ ] [Make sure to `@`mention an existing maintainer to confirm the permissions request, if applicable](https://github.com/jenkins-infra/repository-permissions-updater/#requesting-permissions)
- [x] Use the Jenkins community (LDAP) account name in the YAML file, not the GitHub account name
- [x] [All newly added users have logged in to Artifactory at least once](https://github.com/jenkins-infra/repository-permissions-updater/#requesting-permissions)

### CODE REVIEW

File: permissions/plugin-kubernetes-credentials-provider.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+name: "kubernetes-credentials-provider"
+paths:
+- "com/cloudbees/jenkins/plugins/k8sCredentialProvider"
```
Comment (batmat) [inline permissions/plugin-kubernetes-credentials-provider.yml]: urgh, using not only lowercase is looking for pain IMO... I guess it's too late anyway

**Comment by batmat** [comment]
> For some reason the permissions were not set on artifactory as a result of the hosting process request, so filing manually...

There's no automation for this yet, so this is _normal_.

File: permissions/plugin-kubernetes-credentials-provider.yml (hunk 1 above)
Comment (jtnord) [reply permissions/plugin-kubernetes-credentials-provider.yml]: not tool late there is no release.  
And well even after I can change it as no data is persisted, nor does this constitute a public API.

that said `k8s_credential_provider` would be the convention that is not really any better.

**Comment by daniel-beck** [comment]
This will not work as explained in https://github.com/jenkins-infra/repository-permissions-updater#managing-permissions

**Comment by daniel-beck** [comment]
I don't understand the redundancy in the groupId. Why not just `com.cloudbees.plugins`? https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/blob/master/pom.xml#L35...L36

Also, regarding

> not tool late there is no release.
> And well even after I can change it as no data is persisted, nor does this constitute a public API.

Group IDs are ignored by Jenkins.

**Comment by jtnord** [comment]
> This will not work as explained 

What won't work, I don't see what you are pointing out?

> I don't understand the redundancy in the groupId. Why not just com.cloudbees.plugins? 

because you are assuming that the only thing CloudBees has plugins for is Jenkins, and the only thing we do for Jenkins is plugins...  and we do not want collisions across different maven repos.

>Also, regarding
>> not tool late there is no release.
>> And well even after I can change it as no data is persisted, nor does this constitute a public API.
>
> Group IDs are ignored by Jenkins.


Yeah but `groupId`  *should* match the plugins base `package` in the source code, and I was referring to that.

**Comment by daniel-beck** [comment]
> because you are assuming that the only thing CloudBees has plugins for is Jenkins, and the only thing we do for Jenkins is plugins... and we do not want collisions across different maven repos.

Right, missed that part -- but still `com.cloudbees.jenkins.plugins` makes sense.

> What won't work, I don't see what you are pointing out?

"Paths" is not `groupId`. You are currently allowed to upload files matching `com/cloudbees/jenkins/plugins/k8sCredentialProvider/*/kubernetes-credentials-provider-*`. I doubt this is what you want.

File: permissions/plugin-kubernetes-credentials-provider.yml (hunk 1 above)
Comment (batmat) [reply permissions/plugin-kubernetes-credentials-provider.yml]: do we really even need such a specific groupId? Why not just `com.cloudbees.jenkins.plugins` like Daniel is saying too below?

**Pull request merged**